src/coreComponents/physicsSolvers/multiphysics/PoroelasticSolverEmbeddedFractures.cpp

```diff
@@ -0,0 +1,834 @@
+                                dofManager,
+                                localMatrix,
+                                localRhs );
+
+  // assemble mechanics-flow coupling blocks
+  AssembleCouplingTerms( domain,
+                         dofManager,
+                         localMatrix,
+                         localRhs );
+
+}
+
+void PoroelasticSolverEmbeddedFractures::AssembleCouplingTerms( DomainPartition const & domain,
+                                                                DofManager const & dofManager,
+                                                                CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                                                                arrayView1d< real64 > const & localRhs )
+{
+  GEOSX_MARK_FUNCTION;
+
+  // Rock matrix poroelatic coupling
+  PoroelasticSolver::AssembleCouplingTerms( domain, dofManager, localMatrix, localRhs );
+
+  assembleFractureFlowResidualWrtJump( domain, dofManager, localMatrix, localRhs );
+}
+
+void PoroelasticSolverEmbeddedFractures::
+  assembleTractionBalanceResidualWrtPressure( DomainPartition const & domain,
+                                              DofManager const & dofManager,
+                                              CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                                              arrayView1d< real64 > const & localRhs )
+{
+  GEOSX_MARK_FUNCTION;
+
+  MeshLevel const & mesh                   = *domain.getMeshBody( 0 )->getMeshLevel( 0 );
+  ElementRegionManager const & elemManager = *mesh.getElemManager();
+
+  ConstitutiveManager const * const constitutiveManager = domain.getConstitutiveManager();
+
+  string const jumpDofKey     = dofManager.getKey( SolidMechanicsEmbeddedFractures::viewKeyStruct::dispJumpString );
+  string const pressureDofKey = dofManager.getKey( FlowSolverBase::viewKeyStruct::pressureString );
+
+  ElementRegionManager::MaterialViewAccessor< real64 > const
+  density = elemManager.ConstructFullMaterialViewAccessor< real64 >( "density0",
+                                                                     constitutiveManager );
+  globalIndex const rankOffset = dofManager.rankOffset();
+
+  // Initialise local matrices and vectors
+  array1d< globalIndex >       jumpEqnRowIndices    ( 3 );
+  array1d< globalIndex >       jumpColIndices ( 3 );
+
+  array1d< real64 >            Rfrac( 3 );
+  array1d< real64 >            tractionVec( 3 );
+
+  // Equilibrium matrix
+  array2d< real64 >       eqMatrix( 3, 6 );
+
+  array1d< real64 >       w( 3 );
+
+  real64 dTdpf = 0;
+
+  array2d< real64 > Kwpm_elem( 1, 3 );
+  array2d< real64 > Kwpm_gauss( 1, 3 );
+  array2d< real64 > identityVec( 6, 1 );
+  identityVec.setValues< serialPolicy >( 0 );
+  identityVec[0][0] = 1;
+  identityVec[1][0] = 1;
+  identityVec[2][0] = 1;
+  array1d< real64 > pm( 1 ); // matrix pressure (it's a scalar but let's keep it like this in case we wanted more unknowns)
```

Comment:
You should probably declare all of these as c-arrays inside the loop body (considering we may at some point want to make that loop parallel)

Reply:
yes, as soon as we merge #1147 I ll change these and use `LvArray::tensorOps`.